mitiq/pec/pec.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (C) 2020 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""High-level probabilistic error cancellation tools."""
+
+from typing import Optional, Callable
+import numpy as np
+from mitiq._typing import QPROGRAM
+from mitiq.pec.utils import DecoType
+from mitiq.pec.sampling import sample_circuit
+
+
+def execute_with_pec(
+    circuit: QPROGRAM,
+    executor: Callable[[QPROGRAM], float],
+    deco_dict: DecoType,
+    num_to_average: int = 1,
```

Comment:
Should this be an arg with PEC? I would remove it and keep only `num_samples`.

Reply:
I agree